src/Nethermind/Nethermind.Merge.Plugin/PoSSwitcher.cs

```diff
@@ -118,27 +121,52 @@ private void CheckIfTerminalBlockReached(object? sender, BlockEventArgs e)
 
         private void LoadFinalizedBlockHash()
         {
-            _finalizedBlockHash = LoadHashFromDb(MetadataDbKeys.FinalizedBlockHash) ?? Keccak.Zero;
+            _finalizedBlockHash = _metadataDb.LoadHashFromDb(MetadataDbKeys.FinalizedBlockHash) ?? Keccak.Zero;
+        }
+
+        // Terminal PoW block: A PoW block that satisfies the following conditions pow_block.total_difficulty >= TERMINAL_TOTAL_DIFFICULTY and pow_block.parent_block.total_difficulty < TERMINAL_TOTAL_DIFFICULTY
+        // https://github.com/ethereum/EIPs/blob/d896145678bd65d3eafd8749690c1b5228875c39/EIPS/eip-3675.md#specification
+        public bool IsTerminalBlock(BlockHeader header)
```

Comment:
We're using different function on master now, right?

Reply:
We do?

Reply:
What is the current function?